sdk/tables/azure-data-tables/azure/data/tables/_entity.py

```diff
@@ -3,26 +3,38 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
+import sys
+import datetime
 from enum import Enum
-from typing import Any, Dict, Union, NamedTuple
+from typing import Any, Union, NamedTuple
 
 from azure.core import CaseInsensitiveEnumMeta
 
+if sys.version_info >= (3, 8):
+    from typing import TypedDict  # pylint: disable=no-name-in-module, ungrouped-imports
+else:
+    from typing_extensions import TypedDict
+
+
+class EntityMetadata(TypedDict, total=False):
+    etag: str
+    timestamp: datetime.datetime
```

Comment:
I think this will need to be a bit more verbose to include the fullmetadata (even though we don't return it by default).
We should also document it.
Maybe something like:
```python
from typing_extensions import Required

class EntityMetadata(TypedDict, total=False):
    timestamp: Required[datetime.datetime]
    """A datetime value that is maintained on the server side to record the time an entity was last modified. The Table 
    service uses the Timestamp property internally to provide optimistic concurrency. """
    etag: Required[str]
    """A string representation of the timestamp property. Used to provide optimistic concurrency."""
    id: str
    """The entity ID, which is generally the URL to the resource. This is not returned by default, and only returned if full
    metadata is requested."""
    type: str
    """The type name of the containing object. This is not returned by default, and only returned if full
    metadata is requested."""
    editLink: str
    """The link used to edit/update the entry, if the entity is updatable and the odata.id does not represent a URL
    that can be used to edit the entity. This is not returned by default, and only returned if full metadata is requested."""
    
```